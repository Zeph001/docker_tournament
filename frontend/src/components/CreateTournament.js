import React, { Component } from "react";
import "bootstrap/dist/css/bootstrap.min.css";
import "../css/CreateForm.css";
import {Button, Form} from "react-bootstrap";
import {Link} from "react-router-dom";


export default class CreateTournament extends Component {
  constructor(props) {
    super(props);
    this.state = {
      tournamentName: "",
      bracketSize: "4",
      newTeam: "",
      sentTeams: [],
      singleKO: true,
      doubleKO: false,
      bestOf3: false
    };
    this.handleBracketSelector = this.handleBracketSelector.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
    this.updateTeamsValue = this.updateTeamsValue.bind(this)
    this.updateTournamentName = this.updateTournamentName.bind(this)
    this.handleTournamentType = this.handleTournamentType.bind(this)
    this.handleBestOfMode = this.handleBestOfMode.bind(this)
  }

  handleBracketSelector(event){
    let bracket = event.target.value;
    this.setState({
      bracketSize: bracket
    })
  }

  handleSubmit(e){
   return ( <Link
      to={{
        pathname: "/tournament",
        state: {testState: "test1",
          hideHeader: true,
          receivedTeams: this.state.sentTeams,
          tournamentName: this.state.tournamentName,
          numOfSeeds: this.state.bracketSize,
          singleKO: this.state.singleKO,
          doubleKO: this.state.doubleKO,
          bestOf3: this.state.bestOf3
        }
      }}
    >
    </Link>
   )
  }

updateTeamsValue(e) {
   let arrTeams = this.state.sentTeams
   let idOfInput = parseInt(e.target.getAttribute('id'))
   arrTeams[idOfInput] = e.target.value
   // arrTeams.push(e.target.value)
    this.setState({
      sentTeams: arrTeams
    })
  console.log(arrTeams)
  console.log(this.state.sentTeams)
}

updateTournamentName(e) {
   let newTournamentName = e.target.value
  this.setState({
    tournamentName: newTournamentName
  })
  console.log(this.state.tournamentName)
}

  handleTournamentType(e){
   let tournamentType =  e.target.value
    console.log(tournamentType)
    if(tournamentType ==="Single KO") {
      this.setState({
        singleKO: true,
        doubleKO: false
      })
    } else if(tournamentType ==="Double KO") {
      this.setState({
        singleKO: false,
        doubleKO: true
      })
    }

  }

  handleBestOfMode(e) {
    let bestOfType =  e.target.value
    if(bestOfType ==="Best of 1") {
      this.setState({
        bestOf3: false
      })
    } else if(bestOfType ==="Best of 3") {
      this.setState({
        bestOf3: true
      })
    }
  }
  handleTeamSelection(){
    let bracketSize = this.state.bracketSize

      return(
        <div>
        <div data-testid = "teams" className="input-group">
          <Form.Control data-testid = "team-input1" className ="input-teams" type ="text" id="0" value={this.state.sentTeams[0] || ""}
                        onChange={this.updateTeamsValue} placeholder ="Team 1" required/>
          <Form.Control data-testid = "team-input2"  className ="input-teams" type ="text" id="1" value={this.state.sentTeams[1] || ""}
                        onChange={this.updateTeamsValue} placeholder ="Team 2" required/>
          <Form.Control data-testid = "team-input3" className ="input-teams" type ="text" id="2" value={this.state.sentTeams[2] || ""}
                        onChange={this.updateTeamsValue} placeholder ="Team 3" required/>
          <Form.Control data-testid = "team-input4" className ="input-teams" type ="text" id="3" value={this.state.sentTeams[3] || ""}
                        onChange={this.updateTeamsValue} placeholder ="Team 4" required/>
        </div>

          <div style={{ display: (bracketSize === "4" ? 'none' : 'block') }}>
          <div className="input-group" >
          <Form.Control data-testid = "team-input5" className ="input-teams" type ="text" id="4" value={this.state.sentTeams[4] || ""}
                        onChange={this.updateTeamsValue} placeholder ="Team 5" />
          <Form.Control data-testid = "team-input6" className ="input-teams" type ="text" id="5" value={this.state.sentTeams[5] || ""}
                        onChange={this.updateTeamsValue} placeholder ="Team 6" />
          <Form.Control data-testid = "team-input7" className ="input-teams" type ="text" id="6" value={this.state.sentTeams[6] || ""}
                        onChange={this.updateTeamsValue} placeholder ="Team 7" />
          <Form.Control data-testid = "team-input8" className ="input-teams" type ="text" id="7" value={this.state.sentTeams[7] || ""}
                        onChange={this.updateTeamsValue} placeholder ="Team 8" />
          </div>
          </div>
          <div style={{ display: (bracketSize !== "16" ? 'none' : 'block') }}>
            <div className="input-group">
          <Form.Control data-testid = "team-input9" className ="input-teams" type ="text" id="8" value={this.state.sentTeams[8] || ""}
                        onChange={this.updateTeamsValue} placeholder ="Team 9" />
          <Form.Control data-testid = "team-input10" className ="input-teams" type ="text" id="9" value={this.state.sentTeams[9] || ""}
                        onChange={this.updateTeamsValue} placeholder ="Team 10" />
          <Form.Control data-testid = "team-input11" className ="input-teams" type ="text" id="10" value={this.state.sentTeams[10] || ""}
                        onChange={this.updateTeamsValue} placeholder ="Team 11" />
          <Form.Control data-testid = "team-input12" className ="input-teams" type ="text" id="11" value={this.state.sentTeams[11] || ""}
                        onChange={this.updateTeamsValue} placeholder ="Team 12" />
            </div>
            <div className="input-group">
          <Form.Control data-testid = "team-input13" className ="input-teams" type ="text" id="12" value={this.state.sentTeams[12] || ""}
                        onChange={this.updateTeamsValue} placeholder ="Team 13" />
          <Form.Control data-testid = "team-input14" className ="input-teams" type ="text" id="13" value={this.state.sentTeams[13] || ""}
                        onChange={this.updateTeamsValue} placeholder ="Team 14" />
          <Form.Control data-testid = "team-input15" className ="input-teams" type ="text" id="14" value={this.state.sentTeams[14] || ""}
                        onChange={this.updateTeamsValue} placeholder ="Team 15" />
          <Form.Control data-testid = "team-input16" className ="input-teams" type ="text" id="15" value={this.state.sentTeams[15] || ""}
                        onChange={this.updateTeamsValue} placeholder ="Team 16" />
            </div>
            </div>
        </div>
      )
  }

  render() {
    return (
      <div className="bg-light">
        <div className="container">
          <div className="p-5 text-center">
            <h1 className="mb-3">Create your Tournament</h1>
          </div>
          <Form className ="form-new-team" onSubmit={this.handleSubmit}>
            <Form.Group>
              <Form.Label htmlFor="tournamentNameId" >Name of your Tournament</Form.Label>
              <Form.Control data-testid = "tournament-name" id="tournamentNameId" value={this.state.tournamentName} type="text"
                            placeholder="Tournament Name" onChange ={this.updateTournamentName} required/>
            </Form.Group>
            <Form.Group>
              <Form.Label className="form-label">Tournament Type</Form.Label>
              <div>
                <select data-testid = "select" id="tournament-type" name="tournament-type" className="form-control" onChange={this.handleTournamentType} required>
                  <option data-testid="select-option" value="Single KO">Single KO</option>
                  <option data-testid="select-option" value="Double KO">Double KO</option>
                </select>
              </div>
            </Form.Group>
            <Form.Group>
              <Form.Label className="form-label">Best-Of-Mode</Form.Label>
              <div className="choose-member-level">
                <div className="form-check form-check-inline">
                  <input data-testid="best-of-1-option" name="radio" className="form-check-input" type="radio" id="inlineCheckbox1" value="Best of 1"
                         onChange={this.handleBestOfMode} required/>
                  <label className="form-check-label" htmlFor="inlineCheckbox1">Best of 1</label>
                </div>
                <div className="form-check form-check-inline">
                  <input data-testid="best-of-3-option" name="radio" className="form-check-input" type="radio" id="inlineCheckbox2" value="Best of 3"
                         onChange={this.handleBestOfMode} required/>
                  <label className="form-check-label" htmlFor="inlineCheckbox2">Best of 3</label>
                </div>
              </div>
            </Form.Group>
            <Form.Group>
              <Form.Label className="form-label">Bracket Size</Form.Label>
              <div>
                <select data-testid = "select-bracket" id="tournament-size" name="tournament-size" className="form-control" onChange={this.handleBracketSelector}>
                  <option data-testid = "bracket-option" value="4">4</option>
                  <option data-testid = "bracket-option" value="8">8</option>
                  <option data-testid = "bracket-option" value="16">16</option>
                </select>
              </div>
            </Form.Group>
            <Form.Label htmlFor="teamSelectionId" className="form-label">Insert your Teams</Form.Label>
            <div>
              {this.handleTeamSelection()}
            </div>
            <Form.Group>
              <div className="row form-row">
                <div className="col-2 d-flex justify-content-center">
                    <Button data-testid = "submit-button" type="submit" className="btn btn-primary btn-lg" as = {Link} to = {{
                      pathname: "/tournament",
                      state: {testState: "test1",
                        receivedTeams: this.state.sentTeams,
                        tournamentName: this.state.tournamentName,
                        numOfSeeds: this.state.bracketSize,
                        singleKO: this.state.singleKO,
                        doubleKO: this.state.doubleKO,
                        bestOf3: this.state.bestOf3
                      }
                    }}>Create</Button>

                </div>
              </div>
            </Form.Group>
          </Form>
        </div>
      </div>
    );
  }
}

