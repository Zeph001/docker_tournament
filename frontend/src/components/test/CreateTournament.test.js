import { render, screen, fireEvent } from '@testing-library/react';
import React, { Component } from "react";
import CreateTournament from "../CreateTournament";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {configure, mount, shallow} from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

configure({ adapter: new Adapter() });

let container;
let updateTournamentName;
let wrapper

beforeEach(() => {
  updateTournamentName = CreateTournament.prototype.updateTournamentName
  wrapper = shallow(<CreateTournament/>);
  container = document.createElement('div');
  document.body.appendChild(container);

});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
  jest.clearAllMocks()
});

test("matches the snapshot of CreateTournament", () =>{
  const tree = renderer
    .create(
      <BrowserRouter>
        <CreateTournament/>
      </BrowserRouter>
    )
    .toJSON();
  console.log(tree)
  expect(tree).toMatchSnapshot();
})

const setup = () => {
  const utils = render(<BrowserRouter><CreateTournament /> </BrowserRouter>)
  const tournamentName = utils.getByLabelText('Name of your Tournament')
  const bestOf1 = utils.getByLabelText('Best of 1')
  const bestOf3 = utils.getByLabelText('Best of 3')
  return {
    tournamentName,
    bestOf1,
    bestOf3,
    ...utils,
  }
}
describe("simulates all Tournament Form options", () =>{
  test('simulates Single KO', () =>{
    const { getByTestId, getAllByTestId } = render(<BrowserRouter><CreateTournament /> </BrowserRouter>);
    fireEvent.change(getByTestId('select'), { target: { value: "Single KO" } })
    let options = getAllByTestId('select-option')
    expect(options[0].selected).toBeTruthy();
    expect(options[1].selected).toBeFalsy();
  })

  test('simulates Double KO', () =>{
    const { getByTestId, getAllByTestId } = render(<BrowserRouter><CreateTournament /> </BrowserRouter>);
    fireEvent.change(getByTestId('select'), { target: { value: "Double KO" } })
    let options = getAllByTestId('select-option')
    expect(options[0].selected).toBeFalsy();
    expect(options[1].selected).toBeTruthy();
  })

})

describe('simulates the inputs of Create Tournament', () => {
  test('test the input of Tournament Name', () => {
    const { tournamentName } = setup()
    fireEvent.change(tournamentName, { target: { value: '23' } })
    expect(tournamentName.value).toBe('23')
  })

  test('test the input of Team Name 1', () =>{
    const { getByTestId, getAllByTestId } = render(<BrowserRouter> <CreateTournament /> </BrowserRouter>);
    const contentInput = getByTestId("team-input1");
    fireEvent.change(contentInput, {
      target: { value: "new content" }
    });
    expect(contentInput.value).toEqual("new content")

  })

  test('test the input of Team Name 3', () =>{
    const { getByTestId, getAllByTestId } = render(<BrowserRouter> <CreateTournament /> </BrowserRouter>);
    const contentInput = getByTestId("team-input3");
    fireEvent.change(contentInput, {
      target: { value: "new content" }
    });
    expect(contentInput.value).toEqual("new content")

  })

  test('test the input of Team Name 7', () =>{
    const { getByTestId } = render(<BrowserRouter> <CreateTournament /> </BrowserRouter>);
    const contentInput = getByTestId("team-input7");
    fireEvent.change(contentInput, {
      target: { value: "new content" }
    });
    expect(contentInput.value).toEqual("new content")

  })

  test('test the input of Team Name 13', () =>{
    const { getByTestId } = render(<BrowserRouter> <CreateTournament /> </BrowserRouter>);
    const contentInput = getByTestId("team-input13");
    fireEvent.change(contentInput, {
      target: { value: "new content" }
    });
    expect(contentInput.value).toEqual("new content")

  })
})

describe('Simulates the Best-Of Radio Buttons', () => {
  test('Select Best of 1', () => {
    const {getByTestId} = render(<BrowserRouter> <CreateTournament/> </BrowserRouter>);
    const radio = getByTestId("best-of-1-option");
    fireEvent.change(radio, {
      target: {value: "Best of 1"}
    });
    expect(radio.value).toEqual("Best of 1")
  })
  test('Select Best of 3', () =>{
    const {getByTestId} = render(<BrowserRouter> <CreateTournament/> </BrowserRouter>);
    const radio = getByTestId("best-of-3-option");
    fireEvent.change(radio, {
      target: {value: "Best of 3"}
    });
    expect(radio.value).toEqual("Best of 3")
  })
})

describe("simulates all Bracket Size options", () =>{
  test('Bracket 4', () =>{
    const { getByTestId, getAllByTestId } = render(<BrowserRouter><CreateTournament /> </BrowserRouter>);
    fireEvent.change(getByTestId('select-bracket'), { target: { value: "4" } })
    let options = getAllByTestId('bracket-option')
    expect(options[0].selected).toBeTruthy();
    expect(options[1].selected).toBeFalsy();
    expect(options[2].selected).toBeFalsy();

  })

  test('Bracket 8', () =>{
    const { getByTestId, getAllByTestId } = render(<BrowserRouter><CreateTournament /> </BrowserRouter>);
    fireEvent.change(getByTestId('select-bracket'), { target: { value: "8" } })
    let options = getAllByTestId('bracket-option')
    expect(options[0].selected).toBeFalsy();
    expect(options[1].selected).toBeTruthy();
    expect(options[2].selected).toBeFalsy();

  })

  test('Bracket 16', () =>{
    const { getByTestId, getAllByTestId } = render(<BrowserRouter><CreateTournament /> </BrowserRouter>);
    fireEvent.change(getByTestId('select-bracket'), { target: { value: "16" } })
    let options = getAllByTestId('bracket-option')
    expect(options[0].selected).toBeFalsy();
    expect(options[1].selected).toBeFalsy();
    expect(options[2].selected).toBeTruthy();

  })

})

// describe('simulate the function calls when a input is used', () =>{
//   test('function call of Tournament Name input', () =>{
//     const onChange = jest.fn();
//     const { getByTestId } = render(<BrowserRouter><CreateTournament onChange ={onChange}/></BrowserRouter>);
//
//     fireEvent.change(getByTestId("tournament-name"))
//     expect(onChange).toHaveBeenCalled();
//   })
// })

// it("fills out form and sends it", () =>{
//   const wrapper = mount(<BrowserRouter><CreateTournament/></BrowserRouter>);
//   wrapper.setState({tournamentName: "testName"})
//   console.log(wrapper.find('#tournamentNameId').first().html())
//   const input = wrapper.find('#tournamentNameId')
//   input.instance().value = "test"
//   input.simulate('change')
//   wrapper.update()
//
// })
