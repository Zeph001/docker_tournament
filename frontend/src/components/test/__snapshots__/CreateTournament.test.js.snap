// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot of CreateTournament 1`] = `
<div
  className="bg-light"
>
  <div
    className="container"
  >
    <div
      className="p-5 text-center"
    >
      <h1
        className="mb-3"
      >
        Create your Tournament
      </h1>
    </div>
    <form
      className="form-new-team"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="tournamentNameId"
        >
          Name of your Tournament
        </label>
        <input
          className="form-control"
          data-testid="tournament-name"
          id="tournamentNameId"
          onChange={[Function]}
          placeholder="Tournament Name"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label form-label"
        >
          Tournament Type
        </label>
        <div>
          <select
            className="form-control"
            data-testid="select"
            id="tournament-type"
            name="tournament-type"
            onChange={[Function]}
            required={true}
          >
            <option
              data-testid="select-option"
              value="Single KO"
            >
              Single KO
            </option>
            <option
              data-testid="select-option"
              value="Double KO"
            >
              Double KO
            </option>
          </select>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label form-label"
        >
          Best-Of-Mode
        </label>
        <div
          className="choose-member-level"
        >
          <div
            className="form-check form-check-inline"
          >
            <input
              className="form-check-input"
              data-testid="best-of-1-option"
              id="inlineCheckbox1"
              name="radio"
              onChange={[Function]}
              required={true}
              type="radio"
              value="Best of 1"
            />
            <label
              className="form-check-label"
              htmlFor="inlineCheckbox1"
            >
              Best of 1
            </label>
          </div>
          <div
            className="form-check form-check-inline"
          >
            <input
              className="form-check-input"
              data-testid="best-of-3-option"
              id="inlineCheckbox2"
              name="radio"
              onChange={[Function]}
              required={true}
              type="radio"
              value="Best of 3"
            />
            <label
              className="form-check-label"
              htmlFor="inlineCheckbox2"
            >
              Best of 3
            </label>
          </div>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label form-label"
        >
          Bracket Size
        </label>
        <div>
          <select
            className="form-control"
            data-testid="select-bracket"
            id="tournament-size"
            name="tournament-size"
            onChange={[Function]}
          >
            <option
              data-testid="bracket-option"
              value="4"
            >
              4
            </option>
            <option
              data-testid="bracket-option"
              value="8"
            >
              8
            </option>
            <option
              data-testid="bracket-option"
              value="16"
            >
              16
            </option>
          </select>
        </div>
      </div>
      <label
        className="form-label form-label"
        htmlFor="teamSelectionId"
      >
        Insert your Teams
      </label>
      <div>
        <div>
          <div
            className="input-group"
            data-testid="teams"
          >
            <input
              className="input-teams form-control"
              data-testid="team-input1"
              id="0"
              onChange={[Function]}
              placeholder="Team 1"
              required={true}
              type="text"
              value=""
            />
            <input
              className="input-teams form-control"
              data-testid="team-input2"
              id="1"
              onChange={[Function]}
              placeholder="Team 2"
              required={true}
              type="text"
              value=""
            />
            <input
              className="input-teams form-control"
              data-testid="team-input3"
              id="2"
              onChange={[Function]}
              placeholder="Team 3"
              required={true}
              type="text"
              value=""
            />
            <input
              className="input-teams form-control"
              data-testid="team-input4"
              id="3"
              onChange={[Function]}
              placeholder="Team 4"
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            style={
              Object {
                "display": "none",
              }
            }
          >
            <div
              className="input-group"
            >
              <input
                className="input-teams form-control"
                data-testid="team-input5"
                id="4"
                onChange={[Function]}
                placeholder="Team 5"
                type="text"
                value=""
              />
              <input
                className="input-teams form-control"
                data-testid="team-input6"
                id="5"
                onChange={[Function]}
                placeholder="Team 6"
                type="text"
                value=""
              />
              <input
                className="input-teams form-control"
                data-testid="team-input7"
                id="6"
                onChange={[Function]}
                placeholder="Team 7"
                type="text"
                value=""
              />
              <input
                className="input-teams form-control"
                data-testid="team-input8"
                id="7"
                onChange={[Function]}
                placeholder="Team 8"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            style={
              Object {
                "display": "none",
              }
            }
          >
            <div
              className="input-group"
            >
              <input
                className="input-teams form-control"
                data-testid="team-input9"
                id="8"
                onChange={[Function]}
                placeholder="Team 9"
                type="text"
                value=""
              />
              <input
                className="input-teams form-control"
                data-testid="team-input10"
                id="9"
                onChange={[Function]}
                placeholder="Team 10"
                type="text"
                value=""
              />
              <input
                className="input-teams form-control"
                data-testid="team-input11"
                id="10"
                onChange={[Function]}
                placeholder="Team 11"
                type="text"
                value=""
              />
              <input
                className="input-teams form-control"
                data-testid="team-input12"
                id="11"
                onChange={[Function]}
                placeholder="Team 12"
                type="text"
                value=""
              />
            </div>
            <div
              className="input-group"
            >
              <input
                className="input-teams form-control"
                data-testid="team-input13"
                id="12"
                onChange={[Function]}
                placeholder="Team 13"
                type="text"
                value=""
              />
              <input
                className="input-teams form-control"
                data-testid="team-input14"
                id="13"
                onChange={[Function]}
                placeholder="Team 14"
                type="text"
                value=""
              />
              <input
                className="input-teams form-control"
                data-testid="team-input15"
                id="14"
                onChange={[Function]}
                placeholder="Team 15"
                type="text"
                value=""
              />
              <input
                className="input-teams form-control"
                data-testid="team-input16"
                id="15"
                onChange={[Function]}
                placeholder="Team 16"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="row form-row"
        >
          <div
            className="col-2 d-flex justify-content-center"
          >
            <a
              className="btn btn-primary btn-lg btn btn-primary"
              data-testid="submit-button"
              disabled={false}
              href="/tournament"
              onClick={[Function]}
              type="submit"
            >
              Create
            </a>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
